irc/server.go

```diff
@@ -913,35 +883,30 @@ func (server *Server) setupListeners(config *Config) (err error) {
 	}
 
 	// create new listeners that were not previously configured
-	for _, newaddr := range config.Server.Listen {
-		_, exists := server.listeners[newaddr]
+	numTlsListeners := 0
+	for newAddr, newConfig := range config.Server.trueListeners {
+		if newConfig.TLSConfig != nil {
+			numTlsListeners += 1
+		}
+		_, exists := server.listeners[newAddr]
 		if !exists {
 			// make new listener
-			isTor := isTorListener(newaddr)
-			tlsConfig := tlsListeners[newaddr]
-			listener, listenerErr := server.createListener(newaddr, tlsConfig, isTor, config.Server.UnixBindMode)
+			listener, listenerErr := server.createListener(newAddr, newConfig, config.Server.UnixBindMode)
 			if listenerErr != nil {
-				server.logger.Error("server", "couldn't listen on", newaddr, listenerErr.Error())
+				server.logger.Error("server", "couldn't listen on", newAddr, listenerErr.Error())
 				err = listenerErr
 				continue
 			}
-			server.listeners[newaddr] = listener
-			logListener(newaddr, tlsConfig, isTor)
+			server.listeners[newAddr] = listener
+			logListener(newAddr, newConfig)
 		}
 	}
 
-	if len(tlsListeners) == 0 {
+	if numTlsListeners == 0 {
 		server.logger.Warning("server", "You are not exposing an SSL/TLS listening port. You should expose at least one port (typically 6697) to accept TLS connections")
 	}
 
-	var usesStandardTLSPort bool
-	for addr := range tlsListeners {
-		if strings.HasSuffix(addr, ":6697") {
-			usesStandardTLSPort = true
-			break
-		}
-	}
-	if 0 < len(tlsListeners) && !usesStandardTLSPort {
+	if config.Server.trueListeners[":6697"].TLSConfig == nil {
```

Comment:
Is it just me or would this incorrectly say the "pls expose 6697" message if the admin for example has the listener `"192.168.0.1:6697"` or otherwise specifies a particular IP rather than just the port on all interfaces? If it would, may go back to the method used before this change.